modules/core/src/edu/kit/iti/algover/data/BuiltinSymbols.java

```diff
@@ -210,6 +210,49 @@
                     new FunctionSymbol("$subset", Sort.BOOL, SET1, SET1), 1);
 
 
+    private static final Sort MSET1 = Sort.get("multiset", FunctionSymbolFamily.VAR1);
+
+    public static final FunctionSymbolFamily MULTI_UNION =
+            new FunctionSymbolFamily(
+                    new FunctionSymbol("$multi_union", MSET1, MSET1, MSET1), 1);
+
+    public static final FunctionSymbolFamily MULTI_INTERSECT =
+            new FunctionSymbolFamily(
+                    new FunctionSymbol("$multi_intersect", MSET1, MSET1, MSET1), 1);
+
+    public static final FunctionSymbolFamily MULTI_SET_MINUS =
+            new FunctionSymbolFamily(
+                    new FunctionSymbol("$multi_set_minus", MSET1, MSET1, MSET1), 1);
+
+    public static final FunctionSymbol EMPTY_MULTI_SET =
+            new FunctionSymbol("$multi_empty", Sort.get("multiset", Sort.BOTTOM));
+
+    public static final FunctionSymbolFamily MULTI_CARD =
+            new FunctionSymbolFamily(
+                    new FunctionSymbol("$multi_set_card", Sort.INT, MSET1), 1);
+
+    public static final FunctionSymbolFamily MULTI_SET_ADD =
+            new FunctionSymbolFamily(
+                    new FunctionSymbol("$multi_set_add", MSET1, FunctionSymbolFamily.VAR1, MSET1), 1);
+
+    public static final FunctionSymbolFamily MULTI_SET_IN =
+            new FunctionSymbolFamily(
+                    new FunctionSymbol("$multi_set_in", Sort.BOOL, FunctionSymbolFamily.VAR1, MSET1), 1);
+
+    public static final FunctionSymbolFamily MULTI_SUBSET =
+            new FunctionSymbolFamily(
+                    new FunctionSymbol("$multi_subset", Sort.BOOL, MSET1, MSET1), 1);
+
+    public static final FunctionSymbolFamily SET_TO_MULTI_SET =
+            new FunctionSymbolFamily(
+                    new FunctionSymbol("multiset", MSET1, SET1), 1);
+
+    public static final FunctionSymbolFamily MULTI_SET_ELEM_OCC =
```

Comment:
I added this for support of dafny syntax `ms[elem]` is evaluated to #occurances of `elem` in `ms`.

Reply:
Cool. Is this also implemented in the term translation, and possibly also in the Dafny translation?

Reply:
Yes. Its implemented in TypeResolution, TermBuilder, TreeTermTranslater and a Boogiefier function BoogieVisitor.